k8sclient/k8sclient.go

```diff
@@ -447,11 +467,17 @@ func GetK8sClient(kubeconfig string, kubeClient KubeClient) (KubeClient, error)
 func GetK8sNetwork(k8sclient KubeClient, k8sArgs *types.K8sArgs, confdir string) ([]*types.DelegateNetConf, error) {
 	logging.Debugf("GetK8sNetwork: %v, %v, %v", k8sclient, k8sArgs, confdir)
 
-	netAnnot, defaultNamespace, err := getPodNetworkAnnotation(k8sclient, k8sArgs)
+	netAnnot, defaultNamespace, podID, err := getPodNetworkAnnotation(k8sclient, k8sArgs)
 	if err != nil {
 		return nil, err
 	}
 
+	// Get Pod ComputeDevices info
```

Comment:
Could you please double check that checkpoint.GetComputeDeviceMap() should be called at only when podID != "", correct? If so, please add 'if podID != "" {'...

Reply:
getPodNetworkAnnotation() returns error if it fails to get the Pod from API. I guess its safe to assume we will get a valid podID from v1.Pod struct if getPodNetworkAnnotation() returns error == nil.  Surely we can add some additional check, but I thought that wouldn't be necessary. Let me know you what you think.

Reply:
I mean that this code always lookup checkpoint file (L476). If there is no checkpoint file, i.e. vanilla pod without DP, it always goes to fail with ""GetK8sNetwork: failed to get resourceMap for PodUID", even though network is valid.

I guess this failure has been seen at travis CI testing, hence to avoid regression bug, we have to add it. Could you please add it?

Reply:
Sure. I got it. I am changing the code to initialize resourceMap only when resourceName annotation is present in network CRD. Hopefully this will resolve this issue.